extensions/hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/ReadHadoopNewApiP.java

```diff
@@ -225,6 +233,30 @@ private Supplier(
                             }
                     ).collect(toList());
         }
+
+        private List<IndexedInputSplit> reAssignSplits(List<IndexedInputSplit> assignedSplits) {
+            // If the local file system is marked as shared, no re-assign
+            if (configuration.getBoolean(HadoopSources.SHARED_LOCAL_FS, false)) {
+                return assignedSplits;
+            }
+            try {
+                // If any of the input paths do not belong to LocalFileSystem, no re-assign
+                Job job = Job.getInstance(configuration);
+                Path[] inputPaths = FileInputFormat.getInputPaths(job);
+                for (Path inputPath : inputPaths) {
+                    if (!(inputPath.getFileSystem(configuration) instanceof LocalFileSystem)) {
```

Comment:
Can this ever be `RawLocalFileSystem`?

Reply:
I don't think so but we can add that check too just to be on the safe side